src/ProjectTask.php

```diff
@@ -542,6 +544,10 @@ public function prepareInputForAdd($input)
             }
         }
 
+        if (isset($input['auto_projectstates']) && $input['auto_projectstates'] == true) {
+            $input['projectstates_id'] = self::recalculateStatus($input);
```

Comment:
```suggestion
            $projectstate_id = self::recalculateStatus($input);
            if ($projectstate_id !== false) {
                $input['projectstates_id'] = $projectstate_id;
            }
```
Your new method can return `false`, in which case we shouldn't change the state value.